main.go

```diff
@@ -210,11 +211,18 @@ func main() {
 		log.Error(err, fmt.Sprintf("Couldn't find the cluster domain from the resolv.conf, defaulting to %s", clusterDomain))
 	}
 
+	kubernetesVersion, err := common.GetKubernetesVersion(clientset)
+	if err != nil {
+		log.Error(err, fmt.Sprintf("Unable to resolve Kubernetes version"))
+		os.Exit(1)
```

Comment:
I'm a little leery of exiting on this failure. Did you consider logging and continuing?
I'm not sure if I would suggest defaulting to v1.19 or v1.18. I think v1.18 is probably the safer option.

Reply:
To be more specific. I think if this errors we should not fail and exit but instead log and default to v1.18.